jerry-core/ecma/base/ecma-globals.h

```diff
@@ -1121,6 +1133,22 @@ typedef struct
 
 #endif /* !CONFIG_ECMA_LCACHE_DISABLE */
 
+#ifndef CONFIG_DISABLE_TYPEDARRAY_BUILTIN
+
+/**
+ * Internal properties of TypedArray Object
+ */
+typedef struct
+{
+  /** The arraybuffer object behind the typedarray */
+  ecma_object_t *viewed_arraybuffer_p;
```

Comment:
the length is the typedarray's length, not the arraybuffer's length
for example, 
```
var a = new Int8Array(10); //it will contain an arraybuffer whose length is 10, and its own length is also 10
var b = new int8Array(a.buffer, 2, 5); //b and a will share the same arraybuffer, and b' length is 5
```